src/appleseed-max-impl/appleseedinteractive/appleseedinteractive.cpp

```diff
@@ -0,0 +1,484 @@
+                {
+                    m_renderer->update_camera(m_active_camera);
+                    m_renderer->m_render_session->reininitialize_render();
+                    break;
+                }
+            }
+        }
+
+    private:
+        AppleseedIInteractiveRender*    m_renderer;
+        INode*                          m_active_camera;
+    };
+}
+
+//
+// IInteractiveRender
+//
+
+AppleseedIInteractiveRender::AppleseedIInteractiveRender()
+    : m_owner_wnd(0)
+    , m_currently_rendering(false)
+    , m_bitmap(nullptr)
+    , m_iirender_mgr(nullptr)
+    , m_scene_inode(nullptr)
+    , m_use_view_inode(false)
+    , m_view_inode(nullptr)
+    , m_view_exp(nullptr)
+    , m_progress_cb(nullptr)
+{
+    m_entities.clear();
+}
+
+AppleseedIInteractiveRender::~AppleseedIInteractiveRender(void)
+{
+    // Make sure the active shade session has stopped
+    EndSession();
+}
+
+asf::auto_release_ptr<asr::Project> AppleseedIInteractiveRender::prepare_project(
+    const RendererSettings&     renderer_settings,
+    const ViewParams&           view_params,
+    const TimeValue             time
+)
+{
+    std::string previous_locale(std::setlocale(LC_ALL, "C"));
+
+    RendParams rend_params;
+    rend_params.inMtlEdit = false;
+    rend_params.rendType = RENDTYPE_NORMAL;
+    rend_params.envMap = GetCOREInterface()->GetUseEnvironmentMap() ? GetCOREInterface()->GetEnvironmentMap() : nullptr;
+
+    FrameRendParams frame_rend_params;
+    frame_rend_params.background = Color(GetCOREInterface()->GetBackGround(time, FOREVER));
+    frame_rend_params.regxmin = frame_rend_params.regymin = 0;
+    frame_rend_params.regxmax = frame_rend_params.regymax = 1;
+    
+    // Collect the entities we're interested in.
+    if (m_progress_cb)
+        m_progress_cb->SetTitle(_T("Collecting Entities..."));
+
+    m_entities.clear();
+
+    MaxSceneEntityCollector collector(m_entities);
+    collector.collect(m_scene_inode);
+
+    // Call RenderBegin() on all object instances.
+    render_begin(m_entities.m_objects, time);
+
+    // Build the project.
+    if (m_progress_cb)
+        m_progress_cb->SetTitle(_T("Building Project..."));
+
+    asf::auto_release_ptr<asr::Project> project(
+        build_project(
+            m_entities,
+            m_default_lights,
+            m_view_inode,
+            view_params,
+            rend_params,
+            frame_rend_params,
+            renderer_settings,
+            m_bitmap,
+            time));
+
+    std::setlocale(LC_ALL, previous_locale.c_str());
+
+    return project;
+}
+
+void AppleseedIInteractiveRender::update_camera(INode* camera)
+{
+    ViewParams view_params;
+    get_view_params_from_view_node(view_params, camera, m_time);
+
+    m_project->get_scene()->get_active_camera()->transform_sequence().set_transform(
+        (float)m_time,
+        asf::Transformd::from_local_to_parent(to_matrix4d(Inverse(view_params.affineTM))));
+}
```

Comment:
the method above is called when camera is moved. I'm not sure if it's the correct way to update camera in project.